apps/predictions/lib/stream_supervisor.ex

```diff
@@ -0,0 +1,84 @@
+defmodule Predictions.StreamSupervisor do
+  @moduledoc """
+  DynamicSupervisor managing per-route streams of predictions from the API.
+  """
+
+  use DynamicSupervisor
+
+  alias Routes.Route
+
+  @spec start_link(keyword()) :: Supervisor.on_start()
+  def start_link(opts), do: DynamicSupervisor.start_link(__MODULE__, opts, name: __MODULE__)
+
+  @impl DynamicSupervisor
+  def init(_) do
+    DynamicSupervisor.init(strategy: :one_for_one)
+  end
+
+  @spec ensure_stream_is_started(Route.id_t()) :: {:ok, pid()} | :bypassed
+  def ensure_stream_is_started(route_id),
+    do: ensure_stream_is_started(route_id, System.get_env("USE_SERVER_SENT_EVENTS"))
+
+  defp ensure_stream_is_started(_route_id, "false"), do: :bypassed
```

Comment:
**question:** do I understand correctly that this means that the application will use SSEs unless the `USE_SERVER_SENT_EVENTS` environment variable is explicitly set to `"false"`?

Reply:
Yes, exactly. We happen to use a similar pattern for vehicle data.

Reply:
Okay cool, I was sort of confused by it but as long as it's following an established pattern that's fine.